transport/tchannel/transport.go

```diff
@@ -260,7 +263,9 @@ func (t *Transport) start() error {
 
 		// TODO(abg): If addr was just the port (":4040"), we want to use
 		// ListenIP() + ":4040" rather than just ":4040".
-		listener, err = net.Listen("tcp", addr)
+		lc := net.ListenConfig{}
```

Comment:
right now we are only under `listener == nil` case. What happens `t.listener` is not nil on L250?

Reply:
It will use the `t.listener` as listener. If `t.listener` does not support MPTCP, then the `transport `will not support MPTCP. I added comments in options.go#78. 

MPTCP should be set to true in `ListenConfiguration ` when a listener is created. I think we will not be able to support MPTCP if users passed in a listener in `TransportOptions` unless we implement MPTCP by ourselves.